File: DartLibraryGenerator/scheme_model/api_scheme.dart
import 'api_category.dart';
import 'api_model.dart';

class ApiScheme {
  ApiScheme({
    required this.apiVersion,
    required this.schemeVersion,
    required this.generatedAt,
    required this.categories,
    required this.model,
  });

  final String apiVersion;
  final String schemeVersion;
  final String generatedAt;
  final List<ApiCategory> categories;
  final ApiModel model;

  factory ApiScheme.fromJson(Map<String, dynamic> json) => ApiScheme(
    apiVersion: json["apiVersion"],
    schemeVersion: json["schemeVersion"],
    generatedAt: json["generatedAt"],
    categories: List<ApiCategory>.from(json["categories"].map((x) => ApiCategory.fromJson(x))),
    model: ApiModel.fromJson(json["model"]),
  );

  Map<String, dynamic> toJson() => {
    "apiVersion": apiVersion,
    "schemeVersion": schemeVersion,
    "generatedAt": generatedAt,
    "categories": List<dynamic>.from(categories.map((x) => x.toJson())),
    "model": model.toJson(),
  };
}

File: NetLibraryGenerator/NetLibraryGenerator/SchemeModel/ApiScheme.cs
using Newtonsoft.Json;

namespace NetLibraryGenerator.SchemeModel;

public class ApiScheme
{
    [JsonProperty("apiVersion")] 
    public string ApiVersion { get; set; } = null!;

    [JsonProperty("schemeVersion")]
    public string SchemeVersion { get; set; } = default!;
    
    [JsonProperty("generatedAt")] 
    public string GeneratedAt { get; set; } = null!;

    [JsonProperty("model")]
    public ApiModel Model { get; set; } = null!;
    
    [JsonProperty("categories")] 
    public ApiCategory[] Categories { get; set; } = null!;
}